source/sigv4.c

```diff
@@ -1115,27 +1115,27 @@ static void generateCredentialScope( const SigV4Parameters_t * pSigV4Params,
     /* Each concatenated component is separated by a '/' character. */
     /* Concatenate first 8 characters from the provided ISO 8601 string (YYYYMMDD). */
     ( void ) memcpy( pBufWrite, pSigV4Params->pDateIso8601, ISO_DATE_SCOPE_LEN );
-    pBufWrite += ISO_DATE_SCOPE_LEN;
+    pBufWrite = &pBufWrite[ ISO_DATE_SCOPE_LEN ];
 
     *pBufWrite = CREDENTIAL_SCOPE_SEPARATOR;
-    pBufWrite += CREDENTIAL_SCOPE_SEPARATOR_LEN;
+    pBufWrite = &pBufWrite[ CREDENTIAL_SCOPE_SEPARATOR_LEN ];
 
     /* Concatenate AWS region. */
     ( void ) memcpy( pBufWrite, pSigV4Params->pRegion, pSigV4Params->regionLen );
-    pBufWrite += pSigV4Params->regionLen;
+    pBufWrite = &pBufWrite[ pSigV4Params->regionLen ];
 
     *pBufWrite = CREDENTIAL_SCOPE_SEPARATOR;
-    pBufWrite += CREDENTIAL_SCOPE_SEPARATOR_LEN;
+    pBufWrite = &pBufWrite[ CREDENTIAL_SCOPE_SEPARATOR_LEN ];
 
     /* Concatenate AWS service. */
     ( void ) memcpy( pBufWrite, pSigV4Params->pService, pSigV4Params->serviceLen );
-    pBufWrite += pSigV4Params->serviceLen;
+    pBufWrite = &pBufWrite[ pSigV4Params->serviceLen ];
 
     *pBufWrite = CREDENTIAL_SCOPE_SEPARATOR;
-    pBufWrite += CREDENTIAL_SCOPE_SEPARATOR_LEN;
+    pBufWrite = &pBufWrite[ CREDENTIAL_SCOPE_SEPARATOR_LEN ];
 
     /* Concatenate terminator. */
-    pBufWrite += copyString( pBufWrite, CREDENTIAL_SCOPE_TERMINATOR, CREDENTIAL_SCOPE_TERMINATOR_LEN );
+    pBufWrite = &pBufWrite[ copyString( pBufWrite, CREDENTIAL_SCOPE_TERMINATOR, CREDENTIAL_SCOPE_TERMINATOR_LEN ) ];
```

Comment:
Can we split this into 2 statements?

```suggestion
    size_t result = copyString( pBufWrite, CREDENTIAL_SCOPE_TERMINATOR, CREDENTIAL_SCOPE_TERMINATOR_LEN );
    pBufWrite = &pBufWrite[ result ];
```